docs/source/tutorials_source/package/tutorial_custom_augmentations.py

```diff
@@ -0,0 +1,366 @@
+        mu = sample.mean()
+        snr = np.random.randint(low=4, high=8)
+        sigma = mu / snr
+        noise = torch.normal(torch.zeros(sample.shape), sigma)
+        return sample + noise
+
+
+# %%
+# Now that we have implemented our custom augmentations, we can combine them
+# with available augmentations from the torchvision library. Make sure, that
+# the first augmentation is the histogram normalization, and that the Gaussian
+# noise is applied after converting the image to a tensor.
+#
+# Note that we also transform the image from grayscale to RGB by simply repeating
+# the single color channel three times. The reason for this is that our ResNet expects
+# a three color channel input. This step can be skipped if a different backbone network
+# is used.
+
+
+# compose the custom augmentations with available augmentations
+transform = torchvision.transforms.Compose([
+    HistogramNormalize(),
+    torchvision.transforms.Grayscale(num_output_channels=3),
+    torchvision.transforms.RandomResizedCrop(size=input_size, scale=(0.2, 1.0)),
+    torchvision.transforms.RandomHorizontalFlip(p=0.5),
+    torchvision.transforms.RandomVerticalFlip(p=0.5),
+    torchvision.transforms.GaussianBlur(21),
+    torchvision.transforms.ToTensor(),
+    GaussianNoise(),
+])
+
+
+# %%
+# Let's take a look at what our augmentation pipeline does to an image!
+# We plot the original image on the left and two random augmentations on the 
+# right.
+
+example_image_name = '55e8e3db7309febee415515d06418171.tiff'
+example_image_path = os.path.join(path_to_data, example_image_name)
+example_image = np.array(Image.open(example_image_path))
+
+# torch transform returns a 3 x W x H image, we only show one color channel
+augmented_image_1 = transform(example_image).numpy()[0]
+augmented_image_2 = transform(example_image).numpy()[0]
+
+fig, axs = plt.subplots(1, 3)
+
+axs[0].imshow(example_image)
+axs[0].set_axis_off()
+axs[0].set_title('Original Image')
+
+axs[1].imshow(augmented_image_1)
+axs[1].set_axis_off()
+
+axs[2].imshow(augmented_image_2)
+axs[2].set_axis_off()
+
+# %%
+# Finally, in order to use the augmentation pipeline we defined for self-supervised
+# learning, we need to create a Lightly collate function like so:
+
+# create a collate function which performs the random augmentations
+collate_fn = lightly.data.BaseCollateFunction(transform)
+
+# %%
+# Setup dataset and dataloader
+# ------------------------------
+#
+# We create a dataset which points to the images in the input directory. Since
+# the input images are 16 bits deep, we need to overwrite the image loader such 
+# that it doesn't convert the images to RGB (and hence to 8-bit) automatically.
+
+
+def tiff_loader(f):
+    """Loads a 16-bit tiff image and returns it as a numpy array.
+
+    """
+    with open(f, 'rb') as f:
+        image = Image.open(f)
+        return np.array(image)
+
+# create the dataset and overwrite the image loader
```

Comment:
I would like to get a bit more context here. E.g. mention that lightly follows the torchvision dataset and therefore has a `.loader` property which is called to load an image.
Can you point to the source code to show how the `loader` is used?

The reason is, this is very new. Not everyone looks at the torchvision source code. And even less look at lightly. So we need to explain why there is a loader and why we overwrite it.